File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if (k <= 1) return 0;

        int n = nums.size();
        int low = 0;
        long long product = 1;
        int count = 0;

        for (int high = 0; high < n; high++) {
            product *= nums[high];

            while (product >= k) {
                product /= nums[low];
                low++;
            }

            count += (high - low + 1);
        }

        return count;
    }
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        low = 0
        prod = 1
        count = 0
        high = 0
        if k<= 1:
            return 0
        for i in nums:
            prod = prod * i
            high += 1

            while prod >= k:
                prod = prod // nums[low]
                low += 1
            
            count += high - low 
        
        return count